lib/datadog/appsec/processor.rb

```diff
@@ -39,6 +39,27 @@ def finalize
         end
       end
 
+      class InvalidContextError < StandardError; end
+
+      class << self
+        def current_context
+          Thread.current[:datadog_current_waf_context]
+        end
+
+        def current_context=(context)
+          unless context.instance_of?(Context)
+            raise InvalidContextError,
+              "The context provide: #{context.inspect} is not a Datadog::AppSec::Processor::Context"
+          end
+
+          Thread.current[:datadog_current_waf_context] = context
+        end
+
+        def reset_current_context
```

Comment:
It feels like this should be private.

Reply:
What is the intent of making this method private? I'm not sure having a bunch of send(:method_name) in the codebase makes sense